File: CODECHEF/16_AUG_LONG/2.cpp
#include<iostream>
using namespace std;
int main()
{
	int t,n,temp,i,j,sum,temp1;
	cin>>t;
	while(t--)
	{
		cin>>n;
		i = 0;
		int b[n],check[n];
		while(i<n)
		{
			cin >> temp;
			b[i] = (temp+i+1)%n;
			check[i] = 0;
			i++;
		}
		// cout<<"b "<<b[0]<<b[1]<<b[2]<<b[3]<<endl;
		sum = 0;
		for(i=0;i<n;i++)
		{
			if(check[i]==0)
			{
				j = i;
				while(check[j]==0)
				{
					check[j]+=1;
					j=b[j];
				}
				temp1 = j;
				temp = 1;
				if(check[j]==1)
				{
					check[j]+=1;
					while(b[j]!=temp1)
					{
						j=b[j];
						if(check[j]!=1)
						{
							temp=0;
							break;
						}
						check[j]+=1;
						temp += 1;
					}
					sum += temp;
				}
			}
		}
		cout<<sum<<endl;
	}
}

File: CODECHEF/16_SNACKDOWN/Elimination/2.py
def main():
	t = int(input())
	for ite in range(t):
		x = raw_input()
		n = int(x.split(" ")[0])
		total  = int(x.split(" ")[1])
		if (2*total)%n is not 0:
			print "No"
			continue
		temp = total/n
		check = 0
		if (n%2) is 1:
			if temp > (n-1)/2:
				check = 1
		else:
			if temp > n-1:
				check = 1
			temp1 = (2*total)/n
			k = n/2
			if temp1 >= 4*k:
				check = 1
			if ((temp1 > 2*k) and (temp1%2 is 1)):
				check = 1 

		if check is 1:
			print "Yes"
		else:
			print "No"


main()